plugins/harness-ci-cd/src/hooks/useProjectSlugEntity.tsx

```diff
@@ -86,15 +86,21 @@ export const useProjectSlugFromEntity = (
     }
 
     const serviceUrlParams: any = serviceUrlMatch(selectedPipelineUrl);
+    if (serviceUrlParams) {
+      return {
+        orgId: serviceUrlParams.params.orgId,
+        accountId: serviceUrlParams.params.accountId,
+        serviceId: serviceUrlParams.params.serviceId,
+        urlParams: serviceUrlParams,
+        baseUrl1: baseUrl1,
+        projectIds: serviceUrlParams.params.projectId as string,
+        envFromUrl: envFromUrl,
+      };
+    }
 
+    // Handle the case where neither URL pattern matches
     return {
-      orgId: serviceUrlParams.params.orgId,
-      accountId: serviceUrlParams.params.accountId,
-      serviceId: serviceUrlParams.params.serviceId,
-      urlParams: serviceUrlParams,
-      baseUrl1: baseUrl1,
-      projectIds: serviceUrlParams.params.projectId as string,
-      envFromUrl: envFromUrl,
+      error: 'Invalid URL format',
```

Comment:
Just trying to understand the context a bit more, as it wasn't much clear in the SS, so we are doing error handling for service here?? Whenever there are two annotation one for pipeline and one for service, and the URL in the annotation is wrong for service the plugin will still work and when we navigate to the service page by selecting it from the dropdown it would display this `Invalid IRL Format` error message??

Reply:
> Just trying to understand the context a bit more, as it wasn't much clear in the SS, so we are doing error handling for service here?? Whenever there are two annotation one for pipeline and one for service, and the URL in the annotation is wrong for service the plugin will still work and when we navigate to the service page by selecting it from the dropdown it would display this `Invalid IRL Format` error message??

Hello. Since there wasn't a conditional statement set up for the serviceURL, it assumed that it will always return a match. Because of which when we selected a pipeline with wrong URL, the serviceUrlMatch returned 'undefined'. We've just corrected that so in case both the pipelineUrl and serviceUrl are incorrect, it returns an error message and not just fails the whole page.